### PR TITLE
[aspnetcore] Support cookie parameter

### PR DESCRIPTION
**Pull request by Doxoh**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Support the cookie parameter in aspnetcore

@mandrean 
@jimschubert
@wing328

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache [hunk 1]
```diff
@@ -40,9 +40,15 @@ namespace {{apiPackage}}
...
+        public {{operationModifier}} {{#operationResultTask}}{{#operationIsAsync}}async {{/operationIsAsync}}Task<{{/operationResultTask}}IActionResult{{#operationResultTask}}>{{/operationResultTask}} {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}{{^isCookieParam}}, 
+{{/isCookieParam}}{{/hasMore}}{{/allParams}}){{^generateBody}};{{/generateBody}}
         {{#generateBody}}
-        { {{#responses}}
+        { 
+	    {{#cookieParams}} 
+            var {{paramName}} = Request.Cookies["{{paramName}}"];
+            {{/cookieParams}}
+
+{{#responses}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache]: I would suggest updating line 59 as well as follows:
```
{{/dataType}}
{{/responses}}
```

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache [hunk 2]
```diff
@@ -40,9 +40,15 @@ namespace {{apiPackage}}
...
         [SwaggerResponse(statusCode: {{code}}, type: typeof({{&dataType}}), description: "{{message}}")]{{/dataType}}{{^dataType}}{{/dataType}}{{/responses}}{{/useSwashbuckle}}{{^useSwashbuckle}}{{#responses}}{{#dataType}}
         [ProducesResponseType(statusCode: {{code}}, type: typeof({{&dataType}}))]{{/dataType}}{{^dataType}}{{/dataType}}{{/responses}}{{/useSwashbuckle}}
-        public {{operationModifier}} {{#operationResultTask}}{{#operationIsAsync}}async {{/operationIsAsync}}Task<{{/operationResultTask}}IActionResult{{#operationResultTask}}>{{/operationResultTask}} {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}, {{/hasMore}}{{/allParams}}){{^generateBody}};{{/generateBody}}
+        public {{operationModifier}} {{#operationResultTask}}{{#operationIsAsync}}async {{/operationIsAsync}}Task<{{/operationResultTask}}IActionResult{{#operationResultTask}}>{{/operationResultTask}} {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}{{^isCookieParam}}, 
+{{/isCookieParam}}{{/hasMore}}{{/allParams}}){{^generateBody}};{{/generateBody}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache]: Should this be in just one line (43)?

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache (hunk 1 above)
Comment (Doxoh) [reply modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache]: I have changed it

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache (hunk 2 above)
Comment (Doxoh) [reply modules/openapi-generator/src/main/resources/aspnetcore/2.1/controller.mustache]: I have changed it

**Comment by wing328** [comment]
@Doxoh thanks for the PR, which has been merged into master.

For upcoming changes, please create a new branch for the change.

**Comment by wing328** [comment]
@Doxoh thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553

**Pull request merged**